ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -208,13 +208,15 @@ public Response handleKsqlStatements(
           distributedCmdResponseTimeout);
 
       final List<ParsedStatement> statements = ksqlEngine.parse(request.getKsql());
-      validator.validate(
-          SandboxedServiceContext.create(serviceContext),
-          statements,
-          request.getStreamsProperties(),
-          request.getKsql()
-      );
 
+      if (!request.getKsql().toUpperCase().startsWith("SELECT")) {
```

Comment:
ah this is still needed to avoid httpClientBuilder overhead? Should we consider a better way of special casing pull query? looks like we cannot avoid it. ?